### PR TITLE
[Telink] Turn off  the CHIP_FACTORY_RESET_ERASE_NVS by default

### PR DESCRIPTION
**Pull request by serhiiSalamakha**
Change overview:

- Turn off  the CHIP_FACTORY_RESET_ERASE_NVS by default
- Update the OnFabricRemoved function (AppTaskCommon.cpp)

**Testing**
Tested manually.

### CODE REVIEW

**Comment by serhiiSalamakha** [comment]
@bzbarsky-apple  
Could you perform PR review?

**Review by andy31415 — APPROVED** [review]

File: examples/platform/telink/common/src/AppTaskCommon.cpp [hunk 1]
```diff
@@ -157,51 +157,41 @@ class AppFabricTableDelegate : public FabricTable::Delegate
...
-                Dnssd::CommissioningMode::kEnabledBasic;
             ChipLogProgress(DeviceLayer, "Performing erasing of settings partition");
 
-#ifdef CONFIG_CHIP_FACTORY_RESET_ERASE_NVS
-            void * storage = nullptr;
-            int status     = settings_storage_get(&storage);
-
-            if (status == 0)
+            // Do FactoryReset in case of failed commissioning to allow new pairing via BLE
+            if (sIsCommissioningFailed)
```
Comment (bzbarsky-apple) [inline examples/platform/telink/common/src/AppTaskCommon.cpp:163]: But if PASE did get established and then commissioning fails, won't this test false?

Approving, but it feels like this code is flailing around trying to implement some behavior but not being clear about what that behavior is and how to implement it...

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/platform/telink/common/src/AppTaskCommon.cpp (hunk 1 above)
Comment (serhiiSalamakha) [reply examples/platform/telink/common/src/AppTaskCommon.cpp:163]: I've checked this case. The system works as we expect it to.

We clearly understand how this part should work and what test cases verify it (TC-CADMIN-1.19, TC-OPCREDS-3.6, checking the commissioning fail), but after the previous PR (https://github.com/project-chip/connectedhomeip/pull/28808) it turned out that the previous approach was not obvious, so we decided to update the part (use OnCommissioningSessionEstablishmentError) when we received a request to change CHIP_FACTORY_RESET_ERASE_NVS=n

Comment (serhiiSalamakha) [reply examples/platform/telink/common/src/AppTaskCommon.cpp:163]: @bzbarsky-apple 
If you have no more questions, please resolve the conversation.

Comment (bzbarsky-apple) [reply examples/platform/telink/common/src/AppTaskCommon.cpp:163]: It's your platform, in the end, so as long as this was reviewed by someone else at telink....

**Pull request merged**